src/idl_gen_cpp.cpp

```diff
@@ -90,19 +83,18 @@ struct IDLOptionsCpp : public IDLOptions {
   CppStandard g_cpp_std;    // Base version of C++ standard.
   bool g_only_fixed_enums;  // Generate underlaying type for all enums.
   // clang-format off
-  IDLOptionsCpp(const IDLOptions &opts)
+  IDLOptionsCpp(const IDLOptions& opts)
```

Comment:
Probably you have a wrong code formatter.
I've checked this line again with clang-format, this line in `master` is well formatted according to [GC++CS](https://google.github.io/styleguide/cppguide.html#Reference_Arguments):
```cpp
IDLOptionsCpp(const IDLOptions &opts)
```

I'm not sure but all other code-formatting changes also may be wrong.

Reply:
It's actually this [rule](https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions) which allows either placement of the & or *. Though we should be consistent. 

I am not sure why it changed some and not others, I ran it with Google's internal C++ formatters, I will have to take a look. Running `src/clang-format-git.sh` also didn't find any issues with it.

Reply:
I manually checked the other format changes and didn't notice anything odd. I did see a modification to a clang-format block that I reverted.

I am curious why this particular block is in a //clang-format off block, I don't see the need for special formatting in this case. I am removing this block. I also don't know why the formatter changed this section if it was in the blocks.

Reply:
Our `.clangformat` has:
```
DerivePointerAlignment: false
PointerAlignment: Right
```
so somehow your formatting was ignoring that file?

And yes, that use of `clang-format off` should be removed.

Reply:
Is PointerAlignment: Right not implemented? 
https://stackoverflow.com/questions/38392889/clang-format-align-asterisk-of-pointer-declaration-with-variable-name